instrumentation/netty/netty-4-common/library/src/main/java/io/opentelemetry/instrumentation/netty/v4/common/internal/server/NettyServerInstrumenterFactory.java

```diff
@@ -27,19 +30,23 @@ public static Instrumenter<HttpRequestAndChannel, HttpResponse> create(
       OpenTelemetry openTelemetry,
       String instrumentationName,
       List<String> capturedRequestHeaders,
-      List<String> capturedResponseHeaders) {
+      List<String> capturedResponseHeaders,
+      @Nullable Set<String> knownMethods) {
 
     NettyHttpServerAttributesGetter httpAttributesGetter = new NettyHttpServerAttributesGetter();
 
+    HttpServerAttributesExtractorBuilder<HttpRequestAndChannel, HttpResponse> extractorBuilder =
+        HttpServerAttributesExtractor.builder(
+                httpAttributesGetter, new NettyNetServerAttributesGetter())
+            .setCapturedRequestHeaders(capturedRequestHeaders)
+            .setCapturedResponseHeaders(capturedResponseHeaders);
+    if (knownMethods != null) {
+      extractorBuilder.setKnownMethods(knownMethods);
+    }
```

Comment:
If the builder internally guarded against `null` and just did a noop, these multiple`if` checks sprinkled about could go away. Basically encapsulate this null check.

Reply:
I don't really want to do that -- passing `null`s or using `@Nullable`s on collections (or almost any type that has a "natural" null object value) is a code smell to me. I'm planning to get rid of these horrible nullchecks when I implement https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8779#discussion_r1237096199

Reply:
Legit. 👍🏻